### PR TITLE
Add demo application

### PR DESCRIPTION
**Pull request by kerne1s**
ref: https://app.clickup.com/t/24336023/PRD-1190

### CODE REVIEW

File: libs/mobile/shared/utils/app-env/src/app-env.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { AppEnv } from './env';
+
+export type AppEnvName = 'development' | 'production';
+
+export const appEnv = new AppEnv<AppEnvName>(
+  (process.env.EXPO_PUBLIC_APP_ENV as AppEnvName) ||
+    process.env.NEXT_PUBLIC_APP_ENV,
```
Comment (ekazankova) [inline libs/mobile/shared/utils/app-env/src/app-env.ts:7]: Mobile applications don't use Next enviroment variables

File: libs/mobile/shared/ui/ui-kit/src/lib/components/text-input/component.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    maxLength,
+    onFocus,
+    isStretched = true,
+    multiline,
+    ...props
+  }: AppTextInputProps,
+  ref: React.ForwardedRef<Input>
+): ReactElement {
+  const theme = useAppTheme();
+  const [secured, setSecured] = useState(true);
```
Comment (ekazankova) [inline libs/mobile/shared/ui/ui-kit/src/lib/components/text-input/component.tsx:36]:
```suggestion
  const [isSecured, setIsSecured] = useState(true);
```

File: libs/mobile/shared/ui/ui-kit/src/lib/components/pressable/component.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React, { ReactElement } from 'react';
+import { Pressable, PressableProps, StyleSheet, StyleProp, ViewStyle } from 'react-native';
+
+export interface AppPressableProps extends PressableProps {
+  style?: StyleProp<ViewStyle>;
+}
+
+export function AppPressable({ children, style, ...props }: AppPressableProps): ReactElement {
```
Comment (ekazankova) [inline libs/mobile/shared/ui/ui-kit/src/lib/components/pressable/component.tsx:8]: Can't we use `AppPressable` from `@ronas-it/react-native-common-modules`?

File: libs/mobile/shared/data-access/store/src/store.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { Reducer } from '@reduxjs/toolkit';
+import { setupReactotron } from '@ronas-it/react-native-common-modules/src/data-access/store/utils/reactotron';
```
Comment (ekazankova) [inline libs/mobile/shared/data-access/store/src/store.ts:2]:
```suggestion
import { setupReactotron } from '@ronas-it/react-native-common-modules';
```

File: libs/mobile/shared/data-access/storage/src/service.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import {
+  AsyncStorageItem,
+  SecureStorageItem,
+} from '@ronas-it/react-native-common-modules/src/data-access/storage';
```
Comment (ekazankova) [inline libs/mobile/shared/data-access/storage/src/service.ts:4]:
```suggestion
} from '@ronas-it/react-native-common-modules';
```

File: apps/mobile/app/(main)/_layout.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+export default function MainNavigation(): ReactElement {
+  const translate = useTranslation('APP.MAIN_LAYOUT');
+
+  return (
+    <Tabs backBehavior='none'>
+      <Tabs.Screen
+        name='index'
+        options={{
+          tabBarIcon: ({ focused }) => <Icon name='profile' color={focused ? colors.primary : colors.active} />,
```
Comment (ekazankova) [inline apps/mobile/app/(main)/_layout.tsx:15]: It's better to use colors from the theme (`const theme = useAppTheme()`)

File: libs/mobile/shared/utils/app-env/src/app-env.ts (hunk 1 above)
Comment (kerne1hub) [reply libs/mobile/shared/utils/app-env/src/app-env.ts]: fixed, updated app-env generator

**Comment by kerne1hub** [comment]
@ekazankova please review

File: apps/mobile/app/(main)/_layout.tsx (hunk 6 above)
Comment (ekazankova) [reply apps/mobile/app/(main)/_layout.tsx]: Not changed

**Review by ekazankova — APPROVED** [review]

File: .env
[the remark this answers is not in the document]
Comment (kerne1s) [reply .env:1]: the file is used by NX generators, so without it NX by default will use `eslint.config.js` for libs

**Review by ekazankova — APPROVED** [review]

**Pull request merged**